03-Binary-Search-Trees/04-Binary-Search-Trees-Traversal.md

```diff
@@ -1,12 +1,13 @@
-## Serialization - O(n)
+## Serialization
 
 To serialize a tree, we need to traverse the tree.  This means we need to visit each node and add it to a string or array. There are several ways to do this.
 
-A _traversal_ is an action visiting each node in a graph such as a tree.  There are several kinds of traversals, Breadth First Traversals which visit each node level, by level and Depth First Traversals which visit a node's children before it's siblings.
+A _traversal_ is an action visiting each node in a data structure such as a tree.  There are several kinds of traversals, Breadth First Traversals which visit each node level, by level and Depth First Traversals which visit a node's children before it's siblings.
+
 
 ### Depth First Traversals
 
-Unlike linear data structures like arrays or linked list which have only one logical way to traverse them, trees can be traversed in different ways.  In a _Depth-first traversal_ you explore the children and grandchildren of a node before moving to it's sibling.
+Unlike linear data structures like arrays or linked list which have only one logical way to traverse them, trees can be traversed in different ways.  In a _depth first traversal_ you explore the children, grandchildren, and any further descendants of a node before moving to its sibling and traversing the sibling's descendants.
 
 There are three standard types of depth-first traversals:
```

Comment:
I had a professor who taught me this method for pre-order, post-order, and in-order traversals that has stuck with me over the years. I'm not sure if it's worth adding here but I wanted to mention it: https://blog.devgenius.io/dfs-depth-first-search-traversal-techniques-short-and-sweet-1e4c134babcf

I'm also happy to demonstrate this method in class!

Reply:
Added as a resource! Can definitely go over in class if time